### PR TITLE
Introduce x_DISABLE_COSE_SIGN option

### PR DESCRIPTION
**Pull request by davidvincze**
-- RFC --
Introduce the T_COSE_DISABLE_SIGN option and replace T_COSE_DISABLE_SIGN1 with it.
The tests that use multiple signatures can also be disabled with this option.
COSE_Sign1 structure support is always enabled.

Related issue: #111 

Note:
  The tests are currently failing, but it is expected since this PR and #211 are connected and
  this one is still in progress.

### CODE REVIEW

File: src/t_cose_sign_verify.c [hunk 1]
```diff
@@ -138,6 +139,7 @@ is_soft_verify_error(enum t_cose_err_t error)
 }
 
 
+#ifndef T_COSE_DISABLE_SIGN
```
Comment (laurencelundblade) [inline src/t_cose_sign_verify.c:142]: I think the main disable needed in this file is the code by the comment /* --- The array of COSE_Signatures --- */.  Make the call to process_cose_signatures() conditional (plus the Open/Close array). Return the error from there. The other functions can be conditional too, but even if they aren't -dead_strip will get rid of them.

File: src/t_cose_signature_verify_eddsa.c [hunk 2]
```diff
@@ -176,7 +179,11 @@ t_cose_signature_verify_eddsa_init(struct t_cose_signature_verify_eddsa *me,
 {
     memset(me, 0, sizeof(*me));
     me->s.rs.ident   = RS_IDENT(TYPE_RS_VERIFIER, 'E');
+#ifndef T_COSE_DISABLE_SIGN
     me->s.verify_cb   = t_cose_signature_verify_eddsa_cb;
+#else
+    me->s.verify_cb   = NULL;
```
Comment (laurencelundblade) [inline src/t_cose_signature_verify_eddsa.c:185]: The callbacks are not supposed to be NULL per comment in definition of struct t_cose_signature_verify

File: src/t_cose_signature_verify_main.c [hunk 3]
```diff
@@ -234,13 +235,18 @@ t_cose_signature_verify_main_cb(struct t_cose_signature_verify  *me_x,
...
 
 void
 t_cose_signature_verify_main_init(struct t_cose_signature_verify_main *me)
 {
     memset(me, 0, sizeof(*me));
     me->s.rs.ident = RS_IDENT(TYPE_RS_VERIFIER, 'M');
+#ifndef T_COSE_DISABLE_SIGN
     me->s.verify_cb  = t_cose_signature_verify_main_cb;
+#else
+    me->s.verify_cb  = NULL;
```
Comment (laurencelundblade) [inline src/t_cose_signature_verify_main.c:249]: Should not be NULL

**Review by laurencelundblade — CHANGES_REQUESTED** [review]
This is looking pretty good. Thank you!

T_COSE_DISABLE_COSE_SIGN is a good idea.

I think there is also some code to disable in t_cose_sign_sign.c.

**Comment by laurencelundblade** [comment]
Is DNM De novo mutation? :-)

File: src/t_cose_sign_verify.c (hunk 1 above)
Comment (davidvincze) [reply src/t_cose_sign_verify.c:142]: Updated.

File: src/t_cose_signature_verify_eddsa.c (hunk 2 above)
Comment (davidvincze) [reply src/t_cose_signature_verify_eddsa.c:185]: Updated.

File: src/t_cose_signature_verify_main.c (hunk 3 above)
Comment (davidvincze) [reply src/t_cose_signature_verify_main.c:249]: Updated.

**Comment by davidvincze** [comment]
> Is DNM De novo mutation? :-)

"WIP" might have been better. :)
Thank you for the feedback, I've updated the PR accordingly.

**Pull request merged**